tests/csapi/apidoc_version_test.go

```diff
@@ -11,13 +12,32 @@ import (
 	"github.com/matrix-org/complement/internal/must"
 )
 
+// https://spec.matrix.org/v1.1/#specification-versions
+// altered to limit to X = 1+, as v0 has never existed
+const GlobalVersionRegex = `v[1-9]\d*\.\d+(?:-\S+)?`
+
+// https://github.com/matrix-org/matrix-doc/blob/client_server/r0.6.1/specification/index.rst#specification-versions
+// altered to limit to X = 0 (r0), as r1+ will never exist.
+const r0Regex = `r0\.\d+\.\d+`
+
 func TestVersionStructure(t *testing.T) {
 	deployment := Deploy(t, b.BlueprintAlice)
 	defer deployment.Destroy(t)
-	unauthedClient := deployment.Client(t, "hs1", "")
+
+	client := deployment.Client(t, "hs1", "")
+
 	// sytest: Version responds 200 OK with valid structure
 	t.Run("Version responds 200 OK with valid structure", func(t *testing.T) {
-		res := unauthedClient.MustDo(t, "GET", []string{"_matrix", "client", "versions"}, nil)
+		res := client.MustDoFunc(t, "GET", []string{"_matrix", "client", "versions"})
+
+		// Matches;
+		// - r0.?.?
+		//  where ? is any single digit
+		// - v1^.*(-#)
+		//  where 1^ is 1 through 9 for the first digit, then any digit thereafter,
+		//  and * is any single or multiple of digits
+		//  optionally with dash-separated metadata: (-#)
+		versionRegex, _ := regexp.Compile("^(" + r0Regex + "|" + GlobalVersionRegex + ")$")
```

Comment:
It is common to inline this in the global variable as `var regex = regexp.MustCompile(....)` we should probably just do that and move these lovely comments to there as well.